bidscoin/plugins/pet2bids.py

```diff
@@ -90,7 +92,7 @@ def get_attribute(dataformat: str, sourcefile: Path, attribute: str, options: di
     :return:            The attribute value
     """
 
-    for ext in options['meta']:
+    for ext in options.get('meta', []):
         if sourcefile.with_suffix(ext).is_file():
             LOGGER.warning(f"Reading metadata from an '{ext}' sidecar file is not implemented yet...")
             #  data = pet.helper_functions.single_spreadsheet_reader(sourcefile)
```

Comment:
I think this is where you want to have the spreadsheet reader pull metadata from, but I'm not sure where to edit the `bidsmap_dccn.yaml` to suit.

see:
```yaml
  pet:        # ----------------------- All PET runs ---------------------------
  - provenance:
    attributes: &pet_dicomattr
      Modality: PT
      StudyDescription:
      SeriesDescription:
      SeriesType:
      Radiopharmaceutical:
      ReconstructionMethod:
      ImageType:
      NumberOfSlices:
      NumberOfTimeSlices:
      SliceThickness:
      PixelSpacing:
      Rows:
      Columns:
    bids: &pet_dicoment_pet             # See: schema/datatypes/pet.yaml
      task: <SeriesDescription>
      trc: <Radiopharmaceutical>
      rec: ['', acdyn, acstat, nacdyn, nacstat, 0]
      run: <<1>>
      suffix: pet
    meta: &pet_meta
      TracerName: <<Radiopharmaceutical>>
      TracerRadionuclide:
      InjectedRadioactivity: <<RadionuclideTotalDose>>
      InjectedRadioactivityUnits:
      InjectedMass:
      InjectedMassUnits:
      ModeOfAdministration:
      SpecificRadioactivity:
      SpecificRadioactivityUnits:
      TimeZero: <<RadiopharmaceuticalStartTime>>
      InjectionStart: <<ContrastBolusStartTime>>
      FrameDuration: <<ActualFrameDuration>>
      Units:
  - provenance:
    attributes: *pet_dicomattr
    bids:
      <<: *pet_dicoment_pet
      suffix: events
    meta: *pet_meta
  - provenance:
    attributes: *pet_dicomattr
    bids:
      <<: *pet_dicoment_pet
      recording:
      suffix: blood
    meta: *pet_meta
```

Reply:
`meta` seems a bit overloaded, would adding an `extra data` structure such as with eeg be better for spreadsheet data?

Reply:
> I think this is where you want to have the spreadsheet reader pull metadata from, but I'm not sure where to edit the `bidsmap_dccn.yaml` to suit.

For making the excel attributes plugin specific (which seems very reasonable), yes. If you want to use the metadata in the Excel files to classify source datatypes, all you need to do is to add their key names to the attribute table of (pet) run(s). To have the excel data copied over to the bids folder as json sidecars, there are two ways to do this: 1) Add dynamic values with the excel key names to the meta table 2) implement some code in the `pet2bids.bidscoiner_plugin()` function that reads/writes this data

Reply:
> `meta` seems a bit overloaded, would adding an `extra data` structure such as with eeg be better for spreadsheet data?

If dcm2niix is doing such a bad job that the meta data table gets overloaded, perhaps, but is that really the case? Meta data table complements and/or overrules dcm2niix produced meta data. If the Excel sheet contains a lot of key-value pairs that need to be copied over, then I suggest to always do that -- this is also what is done for the source json sidecars.